### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by crud-engr**

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -5,16 +5,18 @@
   "main": "index.js",
   "type": "module",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "node --experimental-vm-modules node_modules/jest/bin/jest.js",
```
Comment (nawazdhandala) [inline api/package.json:8]: you can directly run `jest`

File: api/test/status.test.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import axios from 'axios';
+
+/**
+ * Test the GET route for home /
+*/
+describe('Dao API', () => {
```
Comment (nawazdhandala) [inline api/test/status.test.js:6]: change this to 'Status API'

**Review by nawazdhandala — CHANGES_REQUESTED** [review]

File: api/package.json (hunk 1 above)
Comment (crud-engr) [reply api/package.json:8]: When I directly run `jest` it's complaining of the import statement.

File: api/test/status.test.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    it('returns status to be OK', async () => {
+        const apiUrl = 'http://localhost:2015/';
```
Comment (nawazdhandala) [inline api/test/status.test.js:8]: You can create a file called utils.js and have the apiurl there. and import it and use it here. So, you dont have to copy paste it every time and if the API url changes we change it in one place.

**Review by nawazdhandala — CHANGES_REQUESTED** [review]

File: api/package.json (hunk 1 above)
Comment (nawazdhandala) [reply api/package.json:8]: Google how to solve it.

File: api/test/status.test.js (hunk 3 above)
Comment (crud-engr) [reply api/test/status.test.js:8]: Noted sir.

File: api/test/status.test.js (hunk 2 above)
Comment (crud-engr) [reply api/test/status.test.js]: Done

File: api/test/status.test.js (hunk 3 above)
Comment (crud-engr) [reply api/test/status.test.js]: Done

File: api/utils/api-url.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const apiConfig = {
```
Comment (nawazdhandala) [inline api/utils/api-url.js:1]: This should be in tests folder. 

also, this should be like 

```
export const API_URL='https://localhost:2015'
```

**Review by nawazdhandala — CHANGES_REQUESTED** [review]

File: api/utils/api-url.js (hunk 4 above)
Comment (crud-engr) [reply api/utils/api-url.js:1]: Alright.

File: web-app/src/tests/App.test.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * PAGE LOADS SUCCESSFULLY
+ */
+describe('On page load', () => {
+    test('Home page loads correctly', async () => {
+        const browser = await puppeteer.launch({});
+        const page = await browser.newPage()
+
+        await page.emulate({
```
Comment (nawazbalto) [inline web-app/src/tests/App.test.js:12]: Remove page.emulate - we dont need this.

File: web-app/src/tests/App.test.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import puppeteer from 'puppeteer';
+import {APP_URL} from './app-url';
+
+/**
+ * PAGE LOADS SUCCESSFULLY
+ */
+describe('On page load', () => {
+    test('Home page loads correctly', async () => {
+        const browser = await puppeteer.launch({});
```
Comment (nawazbalto) [inline web-app/src/tests/App.test.js:9]: You launch and new page in before.

File: web-app/src/tests/App.test.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+            userAgent: ""
+        })
+
+        const app_url = APP_URL;
+
+        await page.goto(app_url);
+        const html = await page.$eval('h5', e => e.innerHTML);
+        expect(html).toBe('Login or Register');
+
+        browser.close();
```
Comment (nawazbalto) [inline web-app/src/tests/App.test.js:26]: You close the browser in after.

File: .github/workflows/api.yaml [hunk 8]
```diff
@@ -0,0 +1,30 @@
+name: API CI
+
+on:
+  push:
+    branches: [ dev ]
```
Comment (nawazbalto) [inline .github/workflows/api.yaml:5]: This should work on all branches and not just dev. Please fix this for the other job as well.

File: web-app/src/tests/App.test.js (hunk 5 above)
Comment (crud-engr) [reply web-app/src/tests/App.test.js:12]: Noted sir.

Comment (crud-engr) [reply web-app/src/tests/App.test.js:12]: Noted sir.

Comment (crud-engr) [reply web-app/src/tests/App.test.js]: Done

File: web-app/src/tests/App.test.js (hunk 6 above)
Comment (crud-engr) [reply web-app/src/tests/App.test.js:9]: I don't understand the review here.

File: web-app/src/tests/App.test.js (hunk 7 above)
Comment (crud-engr) [reply web-app/src/tests/App.test.js:18]: I don't get this also.

File: .github/workflows/api.yaml (hunk 8 above)
Comment (crud-engr) [reply .github/workflows/api.yaml:5]: Alright.

**Review by nawazdhandala — APPROVED** [review]

**Pull request merged**